File: backend/src/main/java/com/gym/management/model/EquipmentMaintenance.java
package com.gym.management.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "equipment_maintenance")
public class EquipmentMaintenance {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @com.fasterxml.jackson.annotation.JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "equipment_id", nullable = false)
    private Equipment equipment;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private MaintenanceType maintenanceType;

    @Column(columnDefinition = "TEXT")
    private String description;

    private String technicianName;
    private String vendor;
    private BigDecimal cost;

    private LocalDate maintenanceDate;
    private LocalDate nextDueDate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private MaintenanceStatus status;

    private String documentUrl;

    @CreationTimestamp
    private LocalDateTime createdAt;

    public enum MaintenanceType {
        PREVENTIVE, REPAIR, INSPECTION
    }

    public enum MaintenanceStatus {
        SCHEDULED, COMPLETED, OVERDUE, CANCELLED
    }
}

File: frontend/src/types/equipmentMaintenance.ts
import type { Equipment } from './equipment';

export interface EquipmentMaintenance {
    id: number;
    equipmentId: number;
    equipment?: Equipment;
    maintenanceType: MaintenanceType;
    description: string;
    technicianName: string;
    vendor?: string;
    cost: number;
    maintenanceDate: string;
    nextDueDate?: string;
    status: MaintenanceStatus;
    documentUrl?: string;
    createdAt?: string;
}

export type MaintenanceType = 'PREVENTIVE' | 'REPAIR' | 'INSPECTION';
export type MaintenanceStatus = 'SCHEDULED' | 'COMPLETED' | 'OVERDUE' | 'CANCELLED';
